### PR TITLE
Simplifies logging configuration.

### PR DESCRIPTION
**Pull request by vladlosev**
@parkr, This simplifies logging configuration so that the tests and samples look cleaner. Also starts using `winston` for logging and adjusts some log event levels in the code.

### CODE REVIEW

File: lib/log.js [hunk 1]
```diff
@@ -1,21 +1,37 @@
...
+    if (this.config.logger) {
+      return this.config.logger;
+    } else {
+      var logger = new winston.Logger({
+        transports: [
+          new winston.transports.Console({
+            level: this.config.level || 'warn',
+            formatter: function(options) {
+              return util.format(
+                '[%s] %s mongodb-fs %s',
```
Comment (parkr) [inline lib/log.js]: One thing I liked about the old log format was that you knew what component the log message was coming from. Would that be possible to add back?

**Comment by parkr** [comment]
This looks great! It would be nice to have the category in the log message to indicate the component that is sending the log message. What do you think?

File: lib/log.js (hunk 1 above)
Comment (vladlosev) [inline lib/log.js]: This wasn't really working before - run the test with the logging enabled (`LOG_LEVEL=trace mocha test/testInProcess.js`) and see all the modules use the same category. :smirk_cat:

**Comment by vladlosev** [comment]
I implemented the component specific categories in the logs. This, however, necessitated reverting back to `log4js` - it has routing and filtering capabilities that are superior to those of `winston`.

**Comment by parkr** [comment]
> This, however, necessitated reverting back to using log4js - it has routing and filtering capabilities that are superior to those of winston.

Do you think `log4js` is inferior to `winston`? I personally prefer it if it makes categories better – it helps to know what component (especially in databases) is doing what. Just a preference.

This PR looks great to me!

**Comment by vladlosev** [comment]
`winston` is much more actively maintained and has more extensive documentation. But as I said,  routing capabilities of `log4js` are better.

**Pull request merged**